server/auth.ts

```diff
@@ -42,6 +43,12 @@ export const authOptions: NextAuthOptions = {
     GitHubProvider({
       clientId: process.env.GITHUB_ID || "",
       clientSecret: process.env.GITHUB_SECRET || "",
+      allowDangerousEmailAccountLinking: true,
```

Comment:
Drive by comment but if you add this it will be pretty easy to hijack someones account on Codu.

All I need to know is your GH email (which is in your commits) and the fact you dont have a gitlab account

Now that I think of it... whats the danger here? I was thinking of creating a Gitlab account with your email but I assume gitlab has an email verification step?

Reply:
<img width="1376" alt="image" src="https://github.com/user-attachments/assets/b860ed5b-259d-4b30-bb1a-787a4dd1be24">

Looks like gitlab has email verification before I can login correctly.

Reply:
Yeah, agreed. Linking is a tricky one! So I think for the moment we can treat them as separate accounts but not allow signups with duplicate email accounts. 

A user can always use their email address to login with @JohnAllenTech's email login anyway. It's been outstanding since last year so we just need to plugin the onboarding flow now and connect all of that.